### ISSUE TITLE
Tuple 

### ISSUE THREAD

**Issue by tchia04** [purcellconsult/Build-Cool-Stuff-With-Python#7]
It seems I can modify a tuple without causing an error.

>>> nums = (1, 3, 5, 7)
>>> print(nums)
(1, 3, 5, 7)
>>> nums = (9,8,7,6)
>>> print(nums)
(9, 8, 7, 6)
>>> nums = (9,8,7)
>>> print(nums)
(9, 8, 7)

**Comment by purcellconsult** [comment]
No, you're doing nothing wrong. Here's some code to help elucidate the issue:

```
>>> nums = (1, 3, 5, 7)
>>> id_1 = id(nums)
>>> nums = (9, 8, 7, 6)
>>> id_2 = id(nums)
>>> id_1 == id_2
False
```

The num variable id changed so this means that the data is being held in two separate addresses in memory.

```
>>> nums = (9, 8, 7)
>>> nums[0] = 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'tuple' object does not support item assignment

```
Item reassignment is not allowed due to immutability. That's why the list structure have functions like append() and pop() and tuple does not.
